src/components/MyRecipes.vue

```diff
@@ -250,14 +356,98 @@ export default {
                 $("#recipe-editor").addClass("show")
                 $("#recipe-editor").removeClass("hide")
             }
-            this.inEditMode = !this.inEditMode
+            this.inEditMode = !this.inEditMode;
         },
         addField(value, field) {
             field.push({ value: { name: "", unit: "", thumbnail: "", link: "" } })
         },
         removeField(index, field) {
-            field.splice(index, 1);
+            if (index > 0) {
+                field.splice(index, 1);
+            }
+        },
+
+
+        selectThumbnail(index, thumbnailUrl) {
+            this.ingredients[index].thumbnail = thumbnailUrl.src.original;
+        },
+
+        async searchIngredientImages(index) {
+            const ingredient = this.ingredients[index].name;
+            const unit = this.ingredients[index].unit;
+            // Check if both name and unit are filled before making the API call
+            if (ingredient && unit) {
+                // Remove the existing imageResults property
+                this.$delete(this.ingredients[index], 'imageResults');
+
+                // Make the API call to fetch the ingredient thumbnails
+                await this.$store.dispatch('fetch_ingredient_thumbnail', ingredient);
+
+                // Retrieve the image results from the store after the action is completed
+                const response = this.$store.state.thumbnail;
+                // Extract the thumbnail URLs from the API response
+                const imageResults = response;
+                // console.log(imageResults[index].url)
+
+
+                // Update the corresponding ingredient object with the image results
+                this.$set(this.ingredients[index], 'imageResults', imageResults);
+                this.$set(this.ingredients[index], 'showImageResults', imageResults.length > 0);
+            }
+        },
+
+        selectThumbnail(index, thumbnailUrl) {
+            this.ingredients[index].thumbnail = thumbnailUrl;
+        },
+
+
+        newPop: function () {
+            $('.button')
+                .popup()
+                ;
+        },
+
+        dropdownValidation: function () {
+            let isValid = true;
+            if (!this.nationality) {
+                this.nationalityError = "The nationality field is required"
+                isValid = false;
+            }
+            if (!this.cookbook_id) {
+                this.cookbookError = "The cookbook field is required";
+                isValid = false
+            }
+            return isValid
+        },
+
+        async submitButton() {
```

Comment:
Why is this asynchronous?

Reply:
I made it async because I need to use the await wait for the API call